src/autora/doc/pipelines/train.py

```diff
@@ -0,0 +1,88 @@
+from typing import Dict, Iterable
+
+import torch
+from datasets import Dataset
+from peft import LoraConfig
+from transformers import AutoModelForCausalLM, AutoTokenizer, TrainingArguments
+from trl import SFTTrainer
+
+from autora.doc.pipelines.data import load_data, preprocess_code
+from autora.doc.runtime.predict_hf import get_quantization_config
+from autora.doc.runtime.prompts import INSTR_SWEETP_1, SYS_GUIDES, PromptBuilder
+
+
+def get_dataset(jsonl_path: str) -> Dataset:
+    # "instruction", "output"
+    inputs, labels = load_data(jsonl_path)
+
+    def gen() -> Iterable[Dict[str, str]]:
+        for i, o in zip(inputs, labels):
+            text = PromptBuilder(SYS_GUIDES, INSTR_SWEETP_1).add_example(preprocess_code(i), o).build()
+            yield {"text": text}
+
+    ds = Dataset.from_generator(gen)
+    return ds
+
+
+def fine_tune(base_model: str, new_model_name: str, dataset: Dataset) -> None:
+    cuda_available = torch.cuda.is_available()
+    # train using 4 bit quantization for lower GPU memory usage
+    kwargs = (
```

Comment:
Below in https://github.com/AutoResearch/autodoc/pull/40/files#diff-0ddceedc62509966b6aa04c9ca82510c5c0577a69568521c39b7e90cf69ad0b7R95 you call this variable `config` here this is kwargs. I was thinking maybe for sake of consistency, this should also be called `config` and you can still use it as `**config` a couple lines down.